### PR TITLE
Plans 2023: Fix grid responsiveness

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to Automattic/martech#1584

## Proposed Changes

Fixes grid responsiveness with 3 (or less) plans.

This does not intend to cleanup any of the complexities with these grids and the table structures underneath. It should be an improvement over production, with columns having a max width of `344px` and wrappers with margins (20px) to not touch the edges. 

### Media

**Before** 

| Before    | After |
| -------- | ------- |
| Screenshot 2023-05-19 at 4 30 54 PM | Screenshot 2023-05-19 at 4 30 23 PM |
| Screenshot 2023-05-19 at 4 39 16 PM | Screenshot 2023-05-19 at 4 37 37 PM |

## Testing Instructions

* Go to each of `/plans/[PAID]`, `/start/plans`, and `/setup/newsletter`
* Confirm the grids render fine across all resolutions (mobile/tablet/desktop).
    * 344px max width of columns
    * 20px grid margin from the edges
    * when on a paid site with credits available, grid should align with the notice bar (see media). This was different in production - grid would extend beyond that point (which looked broken on a 16 inch).
    * grids with 4 plans/columns should now split at 2x2 when in tablet resolution

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by simison** [comment]
I understood from convo (p1683879435888099-slack-C052XEUUBL4) that 3 plans should be 768px wide, now it's 1030px at least in `/setup/newsletter`:

image

**Comment by lucasmendes-design** [comment]
> 3 plans should be 768px wide

I think 768px It's too narrow on big screens. We can have a slightly smaller plans grid since the tailored flows have fewer features, so it's less information to digest. I'm happy to review that if you think it's worthy :) @simison. 

The one I shared is the one below so, It's all good to me.

1600px
image

1440px
image

File: client/my-sites/plan-features-2023-grid/_media-queries.scss
Comment (southp) [inline client/my-sites/plan-features-2023-grid/_media-queries.scss]: Let's remove these commented lines unless there is a special reason :)

File: client/my-sites/plan-features-2023-grid/style.scss
Comment (southp) [inline client/my-sites/plan-features-2023-grid/style.scss]: Same here. Let's remove it unless there is a reason :)

**Comment by simison** [comment]
>I think 768px It's too narrow on big screens. We can have a slightly smaller plans grid since the tailored flows have fewer features, so it's less information to digest. I'm happy to review that if you think it's worthy :) @simison.
>
> The one I shared is the one below so, It's all good to me.

You decide! I was only reviewing per design spec I knew from convo. :-) I don't think exceptions per flow are nice to manage technically, and unnecessarily complicates it. All good to me as well! 👍

**Comment by jeyip** [comment]
## Testing
### Requirements

/plans/[PAID]
- [x] 344px max width of columns
- [x] 20px grid margin from the edges
- [x] when on a paid site with credits available, grid should align with the notice bar (see media). This was different in production - grid would extend beyond that point (which looked broken on a 16 inch).

/start/plans
- [x] 344px max width of columns
- [x] 20px grid margin from the edges
- [x] grids with 4 plans/columns should now split at 2x2 when in tablet resolution ( manually updated visible plans for this p1684861669481249/1684861310.538289-slack-CV2TX2PAN  )

 /setup/newsletter
- [x] 344px max width of columns
- [x] 20px grid margin from the edges

### Browsers
- [x] Edge
- [x] Firefox
- [x] Chrome

**Notes**
> when on a paid site with credits available, grid should align with the notice bar (see media). This was different in production - grid would extend beyond that point (which looked broken on a 16 inch).

I was seeing that the plans _don't_ extend beyond the notice, but the width of the grid doesn't match with the width of the notice in `/plans/[PAID]`. Hoping to confirm that the behavior is what we expect?

Screenshot 2023-05-23 at 9 52 01 AM copy

Screenshot 2023-05-23 at 10 03 32 AM

File: client/my-sites/plan-features-2023-grid/style.scss
Comment (jeyip) [inline client/my-sites/plan-features-2023-grid/style.scss:219]: `NB:`

Is the `$mobile-card-max-width` scss variable necessary? It's only used here for this one selector.

File: client/my-sites/plan-features-2023-grid/style.scss
Comment (jeyip) [inline client/my-sites/plan-features-2023-grid/style.scss]: `NB:`

Should we use `$table-cell-max-width` here?

File: client/my-sites/plan-features-2023-grid/_media-queries.scss
Comment (jeyip) [reply client/my-sites/plan-features-2023-grid/_media-queries.scss]: I second this

**Review by jeyip — COMMENTED** [review]
To me, the plan feature grid layout and styling can be quite complex. I don't have as much experience with the history behind the grid, its breakpoints, its architecture, etc.

From what I can tell through code and through testing, these amendments seem reasonable. +1 when my testing comments and when James's comments are resolved.

File: client/my-sites/plan-features-2023-grid/style.scss
Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/style.scss]: Of course, we could.

File: client/my-sites/plan-features-2023-grid/style.scss
Comment (chriskmnds) [reply client/my-sites/plan-features-2023-grid/style.scss:219]: It makes sense for me to keep this behind a named variable. Irrespective of having multiple uses or not, currently. These styles are hard to navigate as is.

**Review by chriskmnds — COMMENTED** [review]
Thanks for reviewing :-)

I sense folks are reluctant to give this a ✅, yet comments are mostly focused on either minor cleanups (removing a commented-out code - per @southp's review) or preferences (whether a value should live behind a variable or not). Yet all the testing checks out and has had designer approval. I wonder if we are just reluctant to decide between minor and critical concerns or to trust that the author/s will remove a commented-out code after approval? 😕  

This comment I think "justifies" to not approving the PR here (considering everything else in the review):

> I was seeing that the plans don't extend beyond the notice, but the width of the grid doesn't match with the width of the notice in /plans/[PAID]. Hoping to confirm that the behavior is what we expect?

That's because the cells are reaching their maximum width of `344px` with not enough room to render more columns. There will probably always be some difference between table width and notice width (try a grid with 2 or 1 plans and you'll see more of it). I don't really see the issue. 🤔

**Review by chriskmnds — COMMENTED** [review]
Thanks for reviewing :-)

I sense folks are reluctant to give this a ✅, yet comments are mostly focused on either minor cleanups (removing a commented-out code - per @southp's review) or preferences (whether a value should live behind a variable or not). Yet all the testing checks out and has had designer approval. I wonder if we are just reluctant to decide between minor and critical concerns or to trust that the author/s will remove a commented-out code after approval? 😕  

This comment I think "justifies" to not approving the PR here (considering everything else in the review):

> I was seeing that the plans don't extend beyond the notice, but the width of the grid doesn't match with the width of the notice in /plans/[PAID]. Hoping to confirm that the behavior is what we expect?

That's because the cells are reaching their maximum width of `344px` with not enough room to render more columns. There will probably always be some difference between table width and notice width (try a grid with 2 or 1 plans and you'll see more of it). I don't really see the issue. 🤔

**Comment by jeyip** [comment]
> I sense folks are reluctant to give this a ✅, yet comments are mostly focused on either minor cleanups (removing a commented-out code - per @southp's review) or preferences (whether a value should live behind a variable or not). Yet all the testing checks out and has had designer approval.

@chriskmnds 

Not sure if you saw [my comment](https://github.com/Automattic/wp-calypso/pull/77078#pullrequestreview-1440482483), but I said `+1 when my testing comments and when James's comments are resolved.`

Maybe the term "resolved" is misleading? I'm happy to change my wording in the future. What I mean to say is that when we communicate and decide to address **_( or agree not to address )_** my remaining testing comments and James's comments, I think it makes sense to merge 🚢. Follow-ups are totally welcome, as well as alternative suggestions.

> I wonder if we are just reluctant to decide between minor and critical concerns

I think this may also be a miscommunication. I leave `NB` ( non-blocking ) in my comments [here](https://github.com/Automattic/wp-calypso/pull/77078#discussion_r1202929947) and [here](https://github.com/Automattic/wp-calypso/pull/77078#discussion_r1202929009) to denote that these suggestions can be addressed later, or if we decide to, not at all. I think it might be _my_ mistake assuming that it's a common shorthand that folks use at Automattic. I see that you use the phrase `NIT`, so I can switch to that instead.

> or to trust that the author/s will remove a commented-out code after approval? 😕

I definitely trust teammates to handle the work that you're describing, although I think I'm accustomed to different workflows. In my previous team, PR authors would explicitly call out or communicate, in the PR, that they'd be handling certain scopes of work after approvals or in entirely new follow-ups. If things are approached differently on #decepticons, I completely respect that.

For future reference, do folks on the team assume that smaller suggestions that don't receive a response will be handled later? If so, I'm happy to go with the flow. 👍

**Comment by jeyip** [comment]
> the author/s will remove a commented-out code after approval

It sounds like James's suggestions will be handled in a follow up 👍 

> That's because the cells are reaching their maximum width of 344px with not enough room to render more columns. There will probably always be some difference between table width and notice width (try a grid with 2 or 1 plans and you'll see more of it).

Sure that makes sense 👍 

> I don't really see the issue. 🤔

To clarify, in my testing notes I said `Hoping to confirm that the behavior is what we expect?` Maybe I shouldn't keep the comment so short, and I'm happy to add more clarification in the future if i helps.

I'm not intending to convey that it's an issue that needs to be solved. I'm checking in to make sure that the visual and behavioral differences between this PR and production are something that we're aware of, and if not, that we've decided that we're fine with the outcome.

**Review by jeyip — APPROVED** [review]
> From what I can tell through code and through testing, these amendments seem reasonable. +1 when my testing comments and when James's comments are resolved.

We talked through my testing comments and James's comments, so :shipit: 🎉 !

**Comment by chriskmnds** [comment]
Thanks @jeyip. I believe everything (most) addressed  :-)

**Pull request merged**
